Healthchecks introduced for gateway/verifier/btc_indexer/spark_balance_checker

ikripaka opened the pull request:
* healthchecks introduced
* move to 1.90 rust version
* dockerfile fix for btc indexer

ikripaka committed:
add healthcheck for btc_indexer + test + fmt

ikripaka committed:
feat: add health checks implementations for services

feat:
* healthchecks for btc_indexer/gateway/verifier/spark_balance_checker
* add healthcheck in docker-compose.yml

chore:
* implement

fix:
* dockerfile for btc_indexer (issue with migrations)
* remove unused dependencies in spark_client
* change tables dropping sides in btc_indexer
* remove unused configurations in spark_balance_checker

ikripaka committed:
Merge remote-tracking branch 'origin/feature/runes-spark-flow' into feature/healthchecks

# Conflicts:
#	crates/spark_client/Cargo.toml
#	crates/spark_client/src/client.rs
#	gateway/crates/spark_service/src/types.rs
#	gateway/crates/verifier-client/src/deposit_verification_client.rs

kevandee commented on btc_indexer/crates/api/src/api.rs:
```diff
@@ -6,14 +6,14 @@ use ordinals::RuneId;
 impl BtcIndexerApi {
     /// Represents hardcoded `/track_tx` endpoint
     pub const TRACK_TX_ENDPOINT: &'static str = "/track_tx";
+    pub const HEALTHCHECK_ENDPOINT: &'static str = "/healthcheck";
```
/health

kevandee commented on btc_indexer/crates/indexer_internals/src/indexer.rs:
```diff
@@ -115,6 +116,20 @@ impl<C: TitanApi, Db: IndexerDbBounds, TxValidator: TxArbiterTrait> BtcIndexerAp
+                "Threads closed, check internal logic".to_string(),
+            ));
+        }
+        self.persistent_storage
```
Check not only DB readiness, but also indexer_client

kevandee commented on gateway/crates/server/src/handlers/healthcheck.rs:
```diff
@@ -0,0 +1,54 @@
+
+#[instrument(level = "info", skip(state), ret)]
+pub async fn handle(State(state): State<AppState>) -> Result<Json<Empty>, GatewayError> {
+    // TODO: add check of db in 'state.deposit_verification_aggregator'
```
Add a check of the db readiness

kevandee commented on gateway/crates/server/src/init.rs:
```diff
@@ -4,41 +4,70 @@ use axum::routing::post;
+    pub const NOTIFY_RUNES_DEPOSIT_ADDRESS_ENDPOINT: &'static str = "/api/verifier/notify-runes-deposit";
+    pub const BRIDGE_RUNES_ADDRESS_ENDPOINT: &'static str = "/api/user/bridge-runes";
+    pub const TEST_SPARK_ADDRESS_ENDPOINT: &'static str = "/api/test/test-spark";
+    pub const HEALTHCHECK_ENDPOINT: &'static str = "/healthcheck";
```
/health

kevandee commented on spark_balance_checker/crates/server/src/handlers/healthcheck.rs:
```diff
@@ -0,0 +1,21 @@
+    ),
+)]
+pub async fn handle(State(mut state): State<AppState>) -> Result<Json<Empty>, ServerError> {
+    Ok(Json(Empty {}))
```
Check that Spark is ready

ikripaka committed:
Merge remote-tracking branch 'origin/dev' into feature/healthchecks

# Conflicts:
#	btc_indexer/apps/main/src/main.rs
#	gateway/apps/main/src/main.rs
#	gateway/crates/deposit_verification/src/aggregator.rs
#	gateway/crates/deposit_verification/src/error.rs
#	gateway/crates/flow_processor/src/routes/bridge_runes_flow.rs
#	gateway/crates/local_db_store/src/storage.rs
#	gateway/crates/server/src/handlers/mod.rs
#	gateway/crates/server/src/handlers/test_spark.rs
#	gateway/crates/server/src/init.rs
#	gateway/crates/spark_service/src/errors.rs
#	gateway/crates/spark_service/src/types.rs
#	tests/src/gateway_client.rs
#	tests/tests/spark_test.rs
#	verifier/crates/server/src/errors.rs

ikripaka committed:
rename Nonce -> TweakBytes & apply some lints to btc_indexer

generate_nonce -> generate_tweak_bytes
get_address -> get_tweaked_p2tr_address
apply formatting

ikripaka committed:
endpoint rename healthcheck -> health

ikripaka committed:
remove unused weights, changed to previous version spark client auth

ikripaka committed:
remove unused imports, add another trait to simplify healthcheck of verifiers

* remove from gateway redundant typed verifiers creation
* fix healthcheck tests in gateway

ikripaka committed:
remove unused debug trait derive from frost traits

ikripaka committed:
add check for returning status from titan indexer

ikripaka committed:
remove RuneId type created by us

Andezion committed:
fixed tests in indexer_internals and cleaned magic numbers

Andezion committed:
checked tests in local_db_store and fixed magic numbers

ikripaka committed:
remove unused files

remove:
* docker-compose.yml from persistent storage
* assets in btc_indexer/crates/indexer_internal/assets
* unused entries in makefiles
* comments in Dockerfiles

fix:
* error naming in frost (remove redundant string creation)

Andezion committed:
checked tests in server and fixed magic numbers

ikripaka committed:
applied all possible clippy lints

Andezion committed:
cleaned up and fixed tests in frost

Andezion committed:
cleaned up lobal_utils

Andezion committed:
checked proto-hasher tests and fixed magic numbers

Andezion committed:
checked spark-client

Andezion committed:
cleaned token-identifier tests

Andezion committed:
checked flow-processor

ikripaka committed:
introduce healthcheck of spark operator with tonic-health

chore:
* move from dotenv -> dotnevy
* remove unused env_logger

ikripaka committed:
remove ca.pem mentions

ikripaka committed:
Merge branch 'feature/tests_and_magic_numbers' into feature/healthchecks

# Conflicts:
#	btc_indexer/crates/indexer_internals/tests/mock_testing.rs
#	btc_indexer/crates/local_db_store/tests/btc_indexer_requests.rs
#	crates/frost/src/aggregator.rs
#	crates/frost/tests/general.rs
